### PR TITLE
HPCC-13391 Set default for row tag

### PR DESCRIPTION
**Pull request by miguelvazq**
When user try to spray XML file without Row Tag (with empty Row Tag), then the file will sprayed as a delimited (CSV like) file based on the encoding (UTF-xx). Will populate with a value as previously.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

**Review by GordonSmith — APPROVED** [review]
Looks fine

File: esp/src/eclwatch/LZBrowseWidget.js [hunk 1]
```diff
@@ -800,7 +800,7 @@ define([
                     lang.mixin(item, lang.mixin({
                         targetName: item.displayName,
                         targetRecordLength: "",
-                        targetRowTag: "",
+                        targetRowTag: context.i18n.tag,
```
Comment (GordonSmith) [inline esp/src/eclwatch/LZBrowseWidget.js]: Should the default tag be translated?

Comment (richardkchapman) [inline esp/src/eclwatch/LZBrowseWidget.js]: I wouldn't have thought so

Comment (miguelvazq) [inline esp/src/eclwatch/LZBrowseWidget.js]: I agree. I will update.

**Comment by miguelvazq** [comment]
@GordonSmith Updated.

**Review by GordonSmith — APPROVED** [review]
Looks fine - but is there code missing now (or is it on a diff / previous PR)?

**Comment by miguelvazq** [comment]
@GordonSmith In my originally commit I reverted to the original rowtag field value as initially created. I have now added some validation logic to disable spray button. Previously the spray was allowed regardless of field being empty or not.

**Review by GordonSmith — APPROVED** [review]
Looks good.

**Comment by miguelvazq** [comment]
@richardkchapman This is ready to merge.

File: esp/src/eclwatch/LZBrowseWidget.js [hunk 2]
```diff
@@ -800,7 +812,7 @@ define([
                     lang.mixin(item, lang.mixin({
                         targetName: item.displayName,
                         targetRecordLength: "",
-                        targetRowTag: "",
+                        targetRowTag: "tag",
```
Comment (richardkchapman) [inline esp/src/eclwatch/LZBrowseWidget.js]: Is this right, or should it say Row?

**Comment by richardkchapman** [comment]
@miguelvazq One question

File: esp/src/eclwatch/LZBrowseWidget.js (hunk 2 above)
Comment (GordonSmith) [inline esp/src/eclwatch/LZBrowseWidget.js]: Agree - while it doesn't really matter, using "Row" would be a better default (as that is the default tag for an outputted xml dataset).

**Comment by miguelvazq** [comment]
@GordonSmith @richardkchapman Should be good now.

**Pull request merged**